Trimble.Modus.Components/Controls/Toast/TMToastContents.xaml.cs

```diff
@@ -0,0 +1,160 @@
+using Microsoft.Maui.Platform;
+using Trimble.Modus.Components.Enums;
+using Trimble.Modus.Components.Popup.Services;
+
+namespace Trimble.Modus.Components.Controls.Toast;
+
+public partial class TMToastContents : Popup.Pages.PopupPage
+
+{
+    public ImageSource LeftIconSource { get; set; }
+    public string Message { get; set; }
+
+    public string RightIconText { get; set; }
+
+    public ImageSource RightIconSource { get; set; }
+
+    public double ToastWidthRequest { get; set; }
+
+    public Color ToastTheme { get; set; }
+
+    public Color TextColor { get; set; }
+
+
+    PopupNavigation popupNavigation;
+
+    public TMToastContents(ImageSource leftIcon, string message, string rightIconText,Object popupNavigation,ToastTheme theme)
+    {
+        InitializeComponent();
+       
+        setTheme(theme.ToString());
+        this.popupNavigation = (PopupNavigation) popupNavigation;
+        PopupData(leftIcon, message, rightIconText);
+        BindingContext = this;
+        Close();
+    }
+
+    private void setTheme(String theme)
+    {
+        ToastTheme = (Color)BaseComponent.colorsDictionary()[theme];
+        if (string.Equals(theme, "ToastBlue"))
+        {
+            TextColor = (Color)BaseComponent.colorsDictionary()["ToastTextBlue"];
+
+
+        }
+        if (string.Equals(theme, "ToastBlack"))
+        {
+            TextColor = (Color)BaseComponent.colorsDictionary()["ToastWhite"];
+        }
+        else
+        {
+            TextColor = (Color)BaseComponent.colorsDictionary()["ToastBlack"];
+        }
+    }
+
+    public TMToastContents()
+    {
+    }
+
+    private void CloseButton_Clicked(object sender, EventArgs e)
+    {
+         popupNavigation.RemovePageAsync(this, true);
+       
+    }
+    public void Close()
+    {
+        Task.Run(async () => {
+            await Task.Delay(5000);
```

Comment:
Fancy number - Have this as private constant in this class.

Reply:
Added a constant